macos/docs/Controls/Button.md

```diff
@@ -1,40 +1,80 @@
+
+### Format: All of the Above
+In Swift only, the style, size and accentColor can be combined into a single "format" property for convenience.
+```Swift
+let myFormat = ButtonFormat(size: .small, style: .borderless, accentColor: Colors.Palette.blueMagenta30.color)
+let newButton = Button(title: "Button", format: myFormat)
+oldButton.format = myFormat
+```
+
+### Linked Primary Button
+When a Button is pressed, if it has a linkedPrimary Button, it will temporarily apply its format to the linkedPrimary.  This mimics the effect of native macOS secondary buttons taking the accent color highlighting from a nearby primary button when pressed.
 ```Swift
-// There are 3 styles to choose from, the default being primary filled
-let primaryFilledButton = Button(title: "FluentUI Button", style: .primaryFilled),
-let primaryOutlineButton = Button(title: "FluentUI Button", style: .primaryOutline),
-let borderlessButton = Button(title: "FluentUI Button", style: .borderless)
+primaryButton = Button(title: "OK")
+secondaryButton = Button(title: "Cancel", style: .secondary)
+secondaryButton.linkedPrimary = primaryButton
 ```
+![ButtonLinkedPrimary.gif](.attachments/ButtonLinkedPrimary.gif)
 
+### Image and Title
+You can display an image alongside, or instead of, a title.  By default, the image leads the title.
 ```Swift
-// You can display an image instead of a title
-let buttonWithImageAndStyle = Button(image: NSImage(named: NSImage.addTemplateName)!, style: .primaryFilled)
+let buttonWithTitleAndImage = Button(title: "Button", image: NSImage(named: NSImage.addTemplateName)!, imagePosition: .imageTrailing)
+let buttonWithImage = Button(image: NSImage(named: NSImage.addTemplateName)!)
 ```
-### Not recommended
+![ButtonImagePositions.png](.attachments/ButtonImagePositions.png)
+
+## Best Practices
+### Things to do
+- Make sure the label conveys a clear purpose of the button to the user.
+- Describe the action the button will perform, ideally with a verb. Use concise, specific, self-explanatory labels, usually a single word.
+- When using an image in a button, use a template image so that it takes on the style and color of the button.
+- For best results when using the `linkedPrimary` property, the current button should have the `.secondary` style, the linked button should have the `.primary` style, and both buttons should have the same `accentColor`.
+
+### Things to avoid
+- Don't place the default focus on a button that performs a destructive action. Place the default focus on a button that performs a "safe" action instead (e.g "Cancel").
 - Overriding the underlying NSButtonCell will result in undefined behavior. While this may fit your use case, take caution while doing so.
-- Displaying both a title and an image currently does not render properly, and is not recommended.
+- Setting the contentTintColor property will result in undefined behavior.  Use the style and accentColor properties to determine the look of the button instead.
```

Comment:
If we felt strongly about it, we could override the setter for `contentTintColor` to cause a `preconditionFailure()`, which we did in other parts of the button. But I think this being in documentation is good enough.